Bottom: 0070b06e14ed85f53a23906019176805c81acf4c
Top:    d91faed7cfa1260ce7280cf213496861c454159b
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-13 23:23:51 +0000

Fix a few random places where we fail badly if the board outline polygon computes to NULL

Usually, the board outline polygon should default to being the world region, but if a
polygon computation fails, it may return NULL, and we can crash in a few places if that
happens.


---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index a89d4c4..8770c14 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -1692,6 +1692,8 @@ fill_board_outline_holes (hidGC gc, const BoxType *drawn_area)
   struct outline_info info;
 
   ensure_board_outline ();
+  if (PCB->Data->outline == NULL)
+    return;
 
   memset (&polygon, 0, sizeof (polygon));
   polygon.Clipped = PCB->Data->outline;
diff --git a/src/hid/gtk/object3d.c b/src/hid/gtk/object3d.c
index 8578b12..772bfcd 100644
--- a/src/hid/gtk/object3d.c
+++ b/src/hid/gtk/object3d.c
@@ -775,6 +775,9 @@ object3d_from_board_outline (void)
 
   board_outline = board_outline_poly (true);
 
+  if (board_outline == NULL)
+    return NULL;
+
   /* Loop over all board outline pieces */
   pa = board_outline;
   do {
